Graphics/IRenderer.hpp

```diff
@@ -7,42 +7,102 @@
+        M,            ///< The M key
+        N,            ///< The N key
+        O,            ///< The O key
+        P,            ///< The P key
+        Q,            ///< The Q key
+        R,            ///< The R key
+        S,            ///< The S key
+        T,            ///< The T key
+        U,            ///< The U key
+        V,            ///< The V key
+        W,            ///< The W key
+        X,            ///< The X key
+        Y,            ///< The Y key
+        Z,            ///< The Z key
+        Num0,         ///< The 0 key
+        Num1,         ///< The 1 key
+        Num2,         ///< The 2 key
+        Num3,         ///< The 3 key
+        Num4,         ///< The 4 key
+        Num5,         ///< The 5 key
+        Num6,         ///< The 6 key
+        Num7,         ///< The 7 key
+        Num8,         ///< The 8 key
+        Num9,         ///< The 9 key
+        Escape,       ///< The Escape key
+        Space,        ///< The Space key
+        Enter,        ///< The Enter/Return keys
+        Backspace,    ///< The Backspace key
+        Left,         ///< Left arrow
+        Right,        ///< Right arrow
+        Up,           ///< Up arrow
+        Down,         ///< Down arrow
+        F1,           ///< The F1 key
+        F2,           ///< The F2 key
+        F3,           ///< The F3 key
+        F4,           ///< The F4 key
+        F5,           ///< The F5 key
+        F6,           ///< The F6 key
+        F7,           ///< The F7 key
+        F8,           ///< The F8 key
+        F9,           ///< The F9 key
+        F10,          ///< The F10 key
+        F11,          ///< The F11 key
+        F12,          ///< The F12 key
+        Closed        ///< The Closed key
+    } Event;
+
     class IRenderer;
 }
 
 class ArcadeShared::Graphics::IRenderer {
-  public:
-    virtual ~IRenderer() = default;
-
-    /**
-     * @brief Set the Window object of renderer
-     * @param window Renderer's window
-     */
-    virtual void setWindow(IWindow &window) = 0;
-
-    /**
-     * @brief Set the frame rate of renderer
-     * @param fps Renderer frames rate
-     */
-    virtual void setFps(std::size_t fps) = 0;
-
-    /**
-     * @brief Get the window object of renderer
-     * @return Renderer window
-     */
-    virtual IWindow &getWindow() const = 0;
-
-    /**
-     * @brief Get the Fps object of renderer
-     * @return Renderer frames rate
-     */
-    virtual std::size_t getFps() const = 0;
-
-  protected:
-    IWindow     &_window;
-    std::size_t _fps;
+    public:
+      virtual ~IRenderer() = default;
+
+      /**
+       * @brief Get name of renderer
+       */
+      virtual std::string getRendererName() const = 0;
+
+      /**
+       * @brief Get event
+       */
+      virtual ArcadeShared::Graphics::Event getEvent() const = 0;
+
+      /**
+       * @brief Draw Frame
+       */
+      virtual void drawFrame(ArcadeShared::Graphics::IFrame frame) const = 0;
```

Comment:
I don't understand the logic of "Frame" because a frame is a render at a moment but how can you draw it? That have no sense with a basic usage of graphical libs

Reply:
I've changed the name to GraphicsData because this strucutre contains the color for less graphic libraries (libcaca, ncurses, ...) and a path to the texture for more graphic libraries (SFMl, SDL2, ...) and the positions where to display it. Obviously after it the game will return to the Core a list of layers knowing that the layers are GraphicsData lists and the Core will loop on these layers by calling the function "drawGraphicsData"